api/http/proxy/factory/kubernetes/transport.go

```diff
@@ -49,7 +49,61 @@ func (transport *baseTransport) proxyKubernetesRequest(request *http.Request) (*
 	apiVersionRe := regexp.MustCompile(`^(/kubernetes)?/(api|apis/apps)/v[0-9](\.[0-9])?`)
 	requestPath := apiVersionRe.ReplaceAllString(request.URL.Path, "")
 
+	endpointRe := regexp.MustCompile(`([0-9]+)`)
+	endpointIDMatch := endpointRe.FindAllString(request.RequestURI, -1)
```

Comment:
Pass 1 as the second argument will mean it'll match a max of 1.  Since you're only wanting the first one and throw away the rest -1 is for all.